pyomo/contrib/pynumero/algorithms/solvers/scipy_solvers.py

```diff
@@ -289,3 +426,106 @@ def get_pyomo_results(self, model, scipy_results):
         results.solver.number_of_gradient_evaluations = scipy_results.njev
 
         return results
+
+
+class PyomoNewtonSolver(PyomoScipySolver):
+
+    def create_nlp_solver(self, **kwds):
+        nlp = self.get_nlp()
+        solver = NewtonNlpSolver(nlp, **kwds)
+        return solver
+
+    def get_pyomo_results(self, model, scipy_results):
+        nlp = self.get_nlp()
+        results = SolverResults()
+
+        if self._nlp_solver.options.full_output:
+            root, res = scipy_results
+        else:
+            root = scipy_results
+
+        # Record problem data
+        results.problem.name = model.name
+        results.problem.number_of_constraints = nlp.n_eq_constraints()
+        results.problem.number_of_variables = nlp.n_primals()
+        results.problem.number_of_binary_variables = 0
+        results.problem.number_of_integer_variables = 0
+        results.problem.number_of_continuous_variables = nlp.n_primals()
+
+        # Record solver data
+        results.solver.name = "scipy.newton"
+
+        results.solver.wallclock_time = self._timer.timers["solve"].total_time
+
+        if self._nlp_solver.options.full_output:
+            # We only have access to any of this information if the solver was
+            # requested to return its full output.
+
+            # For this solver, res.flag is a string.
+            # If successful, it is 'converged'
+            results.solver.message = res.flag
+
+            if res.converged:
+                term_cond = TerminationCondition.feasible
+            else:
+                term_cond = TerminationCondition.Error
+            results.solver.termination_condition = term_cond
+            results.solver.status = TerminationCondition.to_solver_status(
+                results.solver.termination_condition
+            )
+
+            results.solver.number_of_function_evaluations = res.function_calls
+        return results
+
+
+class PyomoSecantNewtonSolver(PyomoScipySolver):
```

Comment:
Again, this has a lot of repeated code: why not leverage inheritance to reduce copy-and-paste?

Reply:
Done